src/coreclr/jit/gentree.cpp

```diff
@@ -18720,6 +18720,41 @@ bool GenTree::IsArrayAddr(GenTreeArrAddr** pArrAddr)
     return false;
 }
 
+//------------------------------------------------------------------------
+// CanSetZeroFlag: Returns true if this is an arithmetic operation that can
+// set the "zero flag" as part of its operation.
+//
+// Return Value:
+//    True if so.
+//
+// Remarks:
+//    For example, for EQ (AND x y) 0, both xarch and arm64 can emit instructions
+//    that directly set the flags after the 'AND' and thus no comparison is needed.
+//
+bool GenTree::CanSetZeroFlag()
+{
+#if defined(TARGET_XARCH)
+    if (OperIs(GT_AND, GT_OR, GT_XOR, GT_ADD, GT_SUB, GT_NEG))
```

Comment:
This is missing many different cases, particularly those that leave `ZF` in an undefined state such as `imul`

Reply:
(It was factored out of `Lowering::OptimizeConstCompare`)

Reply:
Ah, I see.  I wonder if its going to cause any problematic behavior if we're depending on this to be accurate.

I could easily see someone checking `!CanSetZeroFlag()` and then assuming the zero flag can't be overrwritten. Inversely I could see someone checking `CanSetZeroFlag()` and assuming that it will write some 0/1 value and that it is strictly tied to the computed result.

Reply:
Hmm, I can rename this. The specific meaning I was intending was that this is a node that supports setting the zero flag if `GTF_SET_FLAGS` is set. Maybe `SupportsSettingZeroFlag()` is better?

Reply:
Not sure `Supports` fixes the main concern around people thinking `!Supports` means "does not support" (and therefore will not modify zf).

I don't have a good suggestion for a different name, however.

Reply:
It might just be something we'll have to cover with docs and code review.

Reply:
I renamed it to `SupportsSettingZeroFlag` and added some more docs on it